### DISCUSSION TITLE
Airflow configuration to batch process the DAGS present in AWS S3

### DISCUSSION THREAD

**Discussion by swarupsarangi113** [apache/airflow#27374] [Q&A]
Hey everyone,

Is anyone aware of Airflow config that I can set to basically pick the DAGS from AWS S3 in batches instead of processing them all at once? or do I need to handle it from the python code ?

Example: Let's say I have 300 dags in AWS S3, and instead of picking up all the 300 dags and running them at once, Airflow will pick 30 dags and process them and then next 30 dags and so on ...

**Comment by Taragolis** [comment]
Airflow do not have any integration with third-party storages, Airflow expected that DAG files are stored directly in DAG Folder: https://airflow.apache.org/docs/apache-airflow/stable/concepts/overview.html#architecture-overview.

It could be achieve by different ways:
- GitSync
- Mount shared volume
- Bundle DAGs inside docker image

If you want to limit concurrency/parallelism on different levels you could find very nice how-to in [Astronomer Docs ](https://docs.astronomer.io/learn/airflow-scaling-workers)

If you want to have Cross dependencies between DAGs you've might as well setup this dependencies:
https://airflow.apache.org/docs/apache-airflow/stable/howto/operator/external_task_sensor.html

**Reply by swarupsarangi113** [reply]
Can you explain a bit more? I didn't get your explanation in context with what the question was asked here

**Reply by potiuk** [reply]
I think @swarupsarangi113 it's you whou should explain what you mean by "picking up" . Is it syncinc? parsing? scheduling ? executing ? externl dag executio? I also have no idea. 

@Taragolis explaination touch all of those. But you should explain more what your "picking" is about.

**Reply by swarupsarangi113** [reply]
Currently we are picking airflow dags from AWS S3 bucket and executing them all at once. However I am looking for a custom configuration setting (if present) that will pick the dags from S3 but execute them in batches. Its like if there are 150 dags in total that needs to be executed, airflow will pick all the 150 dags but execute only first 30 and the rest 120 goes to pending state till the first batch is executed.

@potiuk @Taragolis Let me know if my usecase is clear to you now ?

**Reply by potiuk** [reply]
So your rather vague "picking" means really execution - that's what was not clear. look at queues,  pools and see if they can be any good for you. Airlfow does not have a batch mechanism - at most rolling "windows" - i.e. you can have x things being processed at a time (which actually might better for you). Just look https://docs.astronomer.io/learn/airflow-scaling-workers and look for "quues", "pools", "parallelism" in Airflow's docs.
